src/github.com/matrix-org/dendrite/roomserver/input/state.go

```diff
@@ -0,0 +1,258 @@
+package input
+
+import (
+	"fmt"
+	"github.com/matrix-org/dendrite/roomserver/types"
+	"github.com/matrix-org/gomatrixserverlib"
+	"sort"
+)
+
+func calculateAndStoreState(
+	db RoomEventDatabase, event gomatrixserverlib.Event, roomNID types.RoomNID, stateEventIDs []string,
```

Comment:
Comment what `stateEventIDs` are. This is `prev_events`? What state events must be supplied in order for this function to work correctly?